3ds2/src/main/java/com/adyen/checkout/adyen3ds2/Adyen3DS2Configuration.kt

```diff
@@ -106,3 +107,33 @@ class Adyen3DS2Configuration private constructor(
         }
     }
 }
+
+fun CheckoutConfiguration.adyen3DS2(
+    configuration: Adyen3DS2Configuration.Builder.() -> Unit = {}
+): CheckoutConfiguration {
+    val config = Adyen3DS2Configuration.Builder(shopperLocale, environment, clientKey)
+        .apply {
+            amount?.let { setAmount(it) }
+            analyticsConfiguration?.let { setAnalyticsConfiguration(it) }
+        }
+        .apply(configuration)
+        .build()
+    addActionConfiguration(config)
+    return this
+}
+
+fun CheckoutConfiguration.getAdyen3DS2Configuration(): Adyen3DS2Configuration? {
+    return getActionConfiguration(Adyen3DS2Configuration::class.java)
+}
+
+internal fun Adyen3DS2Configuration.toCheckoutConfiguration(): CheckoutConfiguration {
+    return CheckoutConfiguration(
+        shopperLocale = shopperLocale,
+        environment = environment,
+        clientKey = clientKey,
+        amount = amount,
+        analyticsConfiguration = analyticsConfiguration,
+    ) {
+        addActionConfiguration(this@toCheckoutConfiguration)
+    }
+}
```

Comment:
I played around with this, but to do it nicely I have to refactor some things. I propose to do that in a new PR, because this one is already too big. Is that okay?

Reply:
Sure, sounds good!